test/jdk/java/awt/ScrollPane/ScrollPaneWindowsTest/ScrollPaneWindowsTest.java

```diff
@@ -0,0 +1,188 @@
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+  @test
+  @bug 4452612
+  @summary The popup menu of the scroll bar doesn't work properly in Window2000.
+  @key headful
+*/
+
+import java.awt.BorderLayout;
+import java.awt.Dimension;
+import java.awt.EventQueue;
+import java.awt.Frame;
+import java.awt.Insets;
+import java.awt.Panel;
+import java.awt.Robot;
+import java.awt.ScrollPane;
+import java.awt.ScrollPaneAdjustable;
+
+import java.awt.event.AdjustmentListener;
+import java.awt.event.AdjustmentEvent;
+import java.awt.event.InputEvent;
+import java.awt.event.KeyEvent;
+
+import java.util.Properties;
+
+public class ScrollPaneWindowsTest implements AdjustmentListener {
+    ScrollPane sp;
+    Panel p;
+    Robot robot;
+    Frame frame;
+    Insets paneInsets;
+    public static Object LOCK = new Object();
+    ScrollPaneAdjustable vScroll;
+    ScrollPaneAdjustable hScroll;
+    boolean notifyReceived = false;
+
+    public static void main(String[] args) throws Exception {
+        ScrollPaneWindowsTest scrollTest = new ScrollPaneWindowsTest();
+
+        scrollTest.init();
+        scrollTest.start();
+    }
+
+    public void init() throws Exception {
+        EventQueue.invokeAndWait(() -> {
+            frame = new Frame();
+            frame.setLayout(new BorderLayout(1, 1));
+            p = new Panel();
+            p.setLayout(null);
+            p.setSize(new Dimension(800, 800));
+        });
+        robot = new Robot();
+        sp = new ScrollPane(ScrollPane.SCROLLBARS_ALWAYS);
+        vScroll = (ScrollPaneAdjustable) sp.getVAdjustable();
+        hScroll = (ScrollPaneAdjustable) sp.getHAdjustable();
+        vScroll.addAdjustmentListener(this);
+        hScroll.addAdjustmentListener(this);
+
+    }
+
+    public void start() throws Exception {
+        Properties prop = System.getProperties();
+        String os = prop.getProperty("os.name", "").toUpperCase();
+        System.out.println("OS= " + os);
+        if (!os.equals("WINDOWS 2000") && !os.equals("WINDOWS 2003") &&
+                !os.equals("WINDOWS XP")) {
+            System.out.println("This test is for Windows 2000/2003/XP only.");
+            return;
+        }
```

Comment:
Again, use `OSInfo` or better `@requires` tag.

Reply:
The whole comment needs reworking and thinking about.
Is it trying to say this test is obsolete after XP ?
In which case since even the successors of Visa, WIndows 7, 8 and 8.1 are all dead
the comment is really really unwelcome in 2023. And perhaps the test is too.

Reply:
Ok, then I will not move the test, should it be left as applet or it should be removed?

Reply:
probably the idea was to exclude "win 98/etc"? please check for what bug this test was created and if is it applicable for win10 or not.

Reply:
I tested locally as well as in CI system, all the run's were green.